### PR TITLE
public-inbox: init at 1.2.0

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

Supersedes #61158.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @catern

### CODE REVIEW

**Review by teto — APPROVED** [review]
perl packages file doesn't seem generated so you could use ${version} in the urls instead of hardcoding it.

I was able to init and index https://lore.kernel.org/netdev/20191213230022.28144-1-mathew.j.martineau@linux.intel.com/ but didn't test further as I am not familiar with public-inbox yet. Looks ok.

File: pkgs/servers/mail/public-inbox/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    perl certs/create-certs.perl
+  '';
+
+  installTargets = [ "install" ];
+  postInstall = ''
+    for prog in $out/bin/*; do
+        wrapProgram $prog --prefix PATH : ${lib.makeBinPath [ git ]}
+    done
+
+    mv sa_config $sa_config
```
[the remark this answers is not in the document]
Comment (alyssais) [reply pkgs/servers/mail/public-inbox/default.nix]: It stopped building when I did that. I didn’t investigate much — I think it’s easier to leave as it is.

File: pkgs/tools/text/highlight/default.nix [hunk 2]
```diff
@@ -28,10 +28,27 @@ stdenv.mkDerivation rec {
...
+
+  doCheck = true;
+  postCheck = ''
+    perl -Iextras/swig extras/swig/testmod.pl
+  '';
+
+  preInstall = ''
+    mkdir -p $out/${perl.libPrefix}
+    install -m644 extras/swig/highlight.{so,pm} $out/${perl.libPrefix}
+    make -C extras/swig clean # Clean up intermediate files.
```
[the remark this answers is not in the document]
Comment (alyssais) [reply pkgs/tools/text/highlight/default.nix]: You would think so, but [no](https://gitlab.com/saalen/highlight/blob/02e56d68538aeba4ed4216debf5ab60f91413bdd/makefile#L111).

File: pkgs/servers/mail/public-inbox/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (teto) [reply pkgs/servers/mail/public-inbox/default.nix]: yeah I've had trouble with moveToOutput too. Nevermind

**Comment by teto** [comment]
I am of opinon of either merging or disabling darwin, what do you think ?

**Comment by alyssais** [comment]
> I am of opinon of either merging or disabling darwin, what do you think ?

I've "fixed" (by disabling the Perl bindings) highlight on Darwin, since
it worked there before.

As for p-i itself, I can't easily figure out what's going on there
without access to a Darwin machine, and I don't think we should block on
that, so I think this is ready to merge.

**Pull request merged**